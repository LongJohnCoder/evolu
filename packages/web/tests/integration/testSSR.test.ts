// import snapshotDiff from 'snapshot-diff';
import { pageDom, pageUrl, pageAwaitFor50ms } from './helpers';

// emotion changed SSR so it renders different HTML on dev and production.
// TODO: Reenable disabled tests once we figure out how to "fix" it. We
// need `yarn jest` for dev

// test('dom before JS is executed', async () => {
//   await page.goto(pageUrl('testSSR'), { waitUntil: 'domcontentloaded' });
//   await expect(await pageDom()).toMatchSnapshot();
// });

test('dom after JS is executed', async () => {
  await page.goto(pageUrl('testSSR'));
  await pageAwaitFor50ms();
  await expect(await pageDom()).toMatchSnapshot();
});

// test('dom diff before and after JS is executed', async () => {
//   await page.goto(pageUrl('testSSR'), { waitUntil: 'domcontentloaded' });
//   const before = await pageDom();
//   await page.goto(pageUrl('testSSR'));
//   await pageAwaitFor50ms();
//   const after = await pageDom();
//   await expect(snapshotDiff(before, after)).toMatchSnapshot();
// });
