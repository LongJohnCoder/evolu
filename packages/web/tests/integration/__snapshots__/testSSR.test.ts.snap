// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dom after JS is executed 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0 {'hasFocus':false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0 {'hasFocus':false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      tabindex="-1"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;
